GameDB: GSHW fixes for Arc the Lad games.

Tzerinas opened the pull request:
### Description of Changes
Arc the Lad games on PCSX2 are kind of an artifact mess in the emulator, due to what I assume are TC issues (considering Disable Depth Emulation fixes things here).

Exhibits:
![2022-04-01 (1)](https://user-images.githubusercontent.com/98034944/161246090-74f94ad4-8934-4437-91d1-b64a0cf05626.png)
![2022-04-01 (7)](https://user-images.githubusercontent.com/98034944/161246151-44d1b4de-6dd9-45a3-88f6-8a9dcb7183e7.png)

To fully fix this, both DisableDepthEmulation and Skipdraw (to 1) need to be set. Anything else does not work. I did play through Arc a few years ago with these settings and I don't remember anything else broken due to them. (It's been a while though).

On that note, I've seen there's no issue open issue for this one in the issue tracker. Considering how long this issue has been known, I'm wondering if there's a reason for it or if I should make one so it isn't forgotten?

I've also activated Round Sprite (Full) to get rid of various upscaling lines and Half-Pixel Offset: Special (Texture) for Twilight to get rid of a weird *Screen Shake* effect that happens in the main menu when upscaled (Doesn't happen on native or software).

### Rationale behind Changes
Less tinkering. Better experience.

### Suggested Testing Steps
Test the Arc games.

lightningterror reviewed (commented):
Really not in favour of using skipdraw and disabling depth.

Can you post gs dumps?

Tzerinas commented:
Certainly:

https://we.tl/t-cqyJCmIMjs

RedDevilus commented:
Disabling depth emulation on Twilight of Spirits;
Messes with the FMVs:
![image](https://user-images.githubusercontent.com/24227051/162054067-3e11134b-0e9c-4753-b281-07601a1908c7.png)
Messes ingame:
![image](https://user-images.githubusercontent.com/24227051/162054186-3b12601f-6831-433a-854e-ab7b0ed6972f.png)

I suggest you to retest everything to make sure it doesn't make the issues worse than before.

RedDevilus commented:
Some GS dumps:
[ArcTheLadTwilightOfTheSpiritsPAL.zip](https://github.com/PCSX2/pcsx2/files/8430163/ArcTheLadTwilightOfTheSpiritsPAL.zip)
[ArcTheLadTwilightOfTheSpiritsPAL (2).zip](https://github.com/PCSX2/pcsx2/files/8430164/ArcTheLadTwilightOfTheSpiritsPAL.2.zip)

Tzerinas commented:
I.... can't reproduce either of those:

![2022-04-07 (3)](https://user-images.githubusercontent.com/98034944/162083428-52b7807f-5013-4fd3-92c6-a2fc6b7d822c.png)
![2022-04-07 (7)](https://user-images.githubusercontent.com/98034944/162083436-25e3e13a-3de2-4f8e-925e-bbc954cf8f34.png)

Latest master. Both disabledepthemulation and skipdraw. I'm playing on the US version though, so maybe its region specific?

On that note, noob question: How do I play GSDumps?

Tzerinas commented:
On that note, having retested on native I'm not sure Skipdraw is even needed. The issue I noted previously seems to be pronounced on upscaling, but honestly its not bad enough, that the Skipdraw step is needed.

Tzerinas commented:
Removed other regions (since there are at least differences in the PAL version) except US since I can't assure their usefulness and removed Skipdraw (no benefit).

RedDevilus commented:
WX: Debug > Show GS Debugger

Make sure the GSdumps are in snaps:

![image](https://user-images.githubusercontent.com/24227051/162093356-7665173e-8355-4146-8472-4e5add925f75.png)

Qt: Open File > Pick GS Dump file

Tzerinas commented:
Converted to draft. The PAL issue spooked me and I want to make sure there are no regressions. DisableDepth seems to work perfectly for the US version, but I'm now going for a full playthrough first.

lightningterror commented:
Closing as it's not currently worked on and the hw fixes cause issues, when work resumes it can be reopened.
